src/sugar.jl

```diff
@@ -142,14 +142,15 @@ function atset_impl(ex::Expr; overwrite::Bool)
     val = esc(val)
     ret = if ex.head == :(=)
         quote
-            lens = $lens
+            lens = ($lenstransform)($lens)
             $dst = set($obj, lens, $val)
         end
     else
         op = get_update_op(ex.head)
-        f = :(_UpdateOp($op,$val))
+        f = :($_UpdateOp($op,$val))
```

Comment:
I think changes like this need to happen in `parse_obj_lenses` as well. For example, `lens = :($IndexLens($index))`.